pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1226 @@
+
+	if g.MaxItems != 0 {
+		k.MaxItems = &g.MaxItems
+	}
+
+	if g.MinItems != 0 {
+		k.MinItems = &g.MinItems
+	}
+	k.UniqueItems = g.UniqueItems
+
+	if g.MultipleOf != 0 {
+		k.MultipleOf = &g.MultipleOf
+	}
+
+	for _, v := range g.GetEnum() {
+		var convert interface{}
+		if err := v.ToRawInfo().Decode(&convert); err != nil {
+			return err
+		}
+		k.Enum = append(k.Enum, convert)
+	}
+
+	if g.MaxProperties != 0 {
+		k.MaxProperties = &g.MaxProperties
+	}
+
+	if g.MinProperties != 0 {
+		k.MinProperties = &g.MinProperties
+	}
+
+	k.Required = g.Required
+
+	if g.GetItems() != nil {
+		k.Items = &SchemaOrArray{}
+		if len(g.Items.GetSchema()) == 1 {
+			k.Items.Schema = &Schema{}
+			if err := k.Items.Schema.FromGnostic(g.Items.Schema[0]); err != nil {
+				return err
+			}
+		} else {
+			for _, v := range g.Items.Schema {
+				schema := Schema{}
+				if err := schema.FromGnostic(v); err != nil {
+					return nil
+				}
+				k.Items.Schemas = append(k.Items.Schemas, schema)
+			}
+		}
+	}
+
+	for i, v := range g.GetAllOf() {
+		k.AllOf = append(k.AllOf, Schema{})
+		if err := k.AllOf[i].FromGnostic(v); err != nil {
+			return err
+		}
+	}
+
+	if g.Properties != nil {
+		k.Properties = make(map[string]Schema)
+		for _, namedSchema := range g.Properties.AdditionalProperties {
+			val := &Schema{}
+			if err := val.FromGnostic(namedSchema.Value); err != nil {
+				return err
+			}
+
+			k.Properties[namedSchema.Name] = *val
+		}
+	}
+
+	if g.AdditionalProperties != nil {
+		k.AdditionalProperties = &SchemaOrBool{}
+		if g.AdditionalProperties.GetSchema() == nil {
+			k.AdditionalProperties.Allows = g.AdditionalProperties.GetBoolean()
+		} else {
+			k.AdditionalProperties.Schema = &Schema{}
+			if err := k.AdditionalProperties.Schema.FromGnostic(g.AdditionalProperties.GetSchema()); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (k *Schema) FromGnosticSchemaItem(g *openapi_v2.SchemaItem) error {
+	if g == nil {
+		return nil
+	}
+
+	switch p := g.Oneof.(type) {
+	case *openapi_v2.SchemaItem_FileSchema:
+		fileSchema := p.FileSchema
+
+		if err := k.VendorExtensible.FromGnostic(fileSchema.VendorExtension); err != nil {
+			return err
+		}
+
+		k.Format = fileSchema.Format
+		k.Title = fileSchema.Title
+		k.Description = fileSchema.Description
+		k.Required = fileSchema.Required
+		// data loss! fileSchema does not have nullable
+		k.Type = []string{fileSchema.Type}
+		k.ReadOnly = fileSchema.ReadOnly
+
+		if fileSchema.ExternalDocs != nil {
+			k.ExternalDocs = &ExternalDocumentation{}
+			if err := k.ExternalDocs.FromGnostic(fileSchema.ExternalDocs); err != nil {
+				return err
+			}
+		}
+
+		if err := fileSchema.Example.ToRawInfo().Decode(&k.Example); err != nil {
+			return err
+		}
+
+		if err := fileSchema.Default.ToRawInfo().Decode(&k.Default); err != nil {
+			return err
+		}
+
+	case *openapi_v2.SchemaItem_Schema:
+		schema := p.Schema
+
+		if err := k.FromGnostic(schema); err != nil {
+			return nil
+		}
+	}
```

Comment:
need a default case that either returns an error or a boolean for data loss

Reply:
this was actually a mistake to `return nil`. I meant to `return err`. In any case, the boolean has been added, and the erroneous return corrected.